### DISCUSSION TITLE
Siri control (esp32+mic+speaker+button)

### DISCUSSION THREAD

**Discussion by luke2023** [pschatzmann/ESP32-A2DP#363] [Q&A] [answered]
Hello, do you have any idea of how to create a siri control example to (act like a AirPod)
Volume up/down
Call siri
Receive iPhone sound and send it to speaker
Send mic signal to iPhone

Thank you very much

LinChe

**Comment by pschatzmann** [comment, marked as answer]
I am pretty sure that Apple uses some (secret) custom protocol to do this.
So I expect that this is just not possible to integrate Siri!

**Comment by mason105** [comment]
https://www.bluetooth.com/wp-content/uploads/attachments/BluetoothDesignGuidelines.pdf   
There is some detail on chap 15.is it possible do it on esp32

**Comment by mason105** [comment]
15.1 Enabling Custom Siri Commands
Every accessory that supports Siri over Bluetooth must support HFP Command AT+XAPL (page 78). The device
will use the information sent by this command to enable and disable custom commands related to Siri.
To receive Siri status events, the accessory must send the AT+XAPL command after making a successful HFP
Service Level Connection (SLC) to the device. The accessory should send an AT+XAPL command first, before
sending any of the additional Siri-specific commands described below....
